lib/MailNotifications.php

```diff
@@ -0,0 +1,412 @@
+				//  but it would likely be a better choice to have this stored in a db like the activity mail queue?
+				$this->config->setUserValue($uid, 'notifications', 'mail_last_send_id', $lastSendId);
+				$this->config->setUserValue($uid, 'notifications', 'mail_last_send_time', $now);
+				$this->config->setUserValue($uid, 'notifications', 'mail_last_check_was_instant', $isInstantNotify ? 1 : 0);
+			} catch (\Exception $e) {
+				$this->logger->error($e->getMessage());
+			}
+		}
+	}
+
+	/**
+	 * prepare the contents of the email message containing the provided list of notifications
+	 *
+	 * @param string $uid
+	 * @param INotificaiton[] $notifications
+	 * @param string $language
+	 * @return IMessage message contents
+	 */
+	protected function prepareEmailMessage(string $uid, array $notifications, string $language): IMessage {
+		$user = $this->userManager->get($uid);
+		$userEmailAddress = $user->getEMailAddress();
+
+		if (empty($userEmailAddress)) {
+			return null;
+		}
+
+		// Prepare our email template
+		$l10n = $this->l10nFactory->get('notifications', $language);
+
+		$template = $this->mailer->createEMailTemplate('notifications.EmailNotification', [
+			'displayname' => $user->getDisplayName(),
+			'url' => $this->urlGenerator->getAbsoluteURL('/')
+		]);
+
+		// Prepare email header
+		$template->addHeader();
+		$template->addHeading($l10n->t('Hello %s',[$user->getDisplayName()]), $l10n->t('Hello %s,',[$user->getDisplayName()]));
+
+		// Prepare email subject and body mentioning amount of notifications
+		$homeLink = '<a href="' . $this->urlGenerator->getAbsoluteURL('/') . '">' . htmlspecialchars($this->defaults->getName()) . '</a>';
+		$notificationsCount = count($notifications);
+		$template->setSubject($l10n->n('New notification for %s', '%n new notifications for %s', $notificationsCount, [$this->defaults->getName()]));
+		$template->addBodyText(
+			$l10n->n('You have a new notification for %s', 'You have %n new notifications for %s', $notificationsCount, [$homeLink]),
+			$l10n->n('You have a new notification for %s', 'You have %n new notifications for %s', $notificationsCount, [$this->urlGenerator->getAbsoluteURL('/')])
+		);
+
+		// Prepare email body with the content of missed notifications
+		// Notifications are assumed to be passed in in descending order (latest first). Reversing to present chronologically.
+		$notifications = array_reverse($notifications);
+
+		$timezone = $this->config->getUserValue($uid, 'core', 'timezone', date_default_timezone_get());
+
+		foreach ($notifications as $notification) {
+			$relativeDateTime = $this->dateFormatter->formatDateTimeRelativeDay($notification->getDateTime(), 'long', 'short', new \DateTimeZone($timezone), $l10n);
+			$template->addBodyListItem($this->getHTMLContents($notification), $relativeDateTime, $notification->getIcon(), $notification->getParsedSubject());
+
+			// Buttons probably were not intended for this, but it works ok enough for showing the idea.
```

Comment:
Only for the sample you picked.
Actions have different request types and you should only show those with request type WEB:
https://github.com/nextcloud/server/blob/215aef3cbdc1963be1bb6bca5218ee0a4b7f1665/lib/public/Notification/IAction.php#L50-L53

Others will mostlikely always result in a 404 or another error when being opened in the browser.

Reply:
Alright, I filtered these only